### PR TITLE
[CN-457] Add Wan Replication doc

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

**Review by dzeromski-hazelcast — APPROVED** [review]

**Review by hasancelik — APPROVED** [review]

**Comment by hasancelik** [comment]
@JakeSCahill could you review this PR?

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+----
+kubectl get wanreplication wanreplication-sample -o wide
+----
+
+----
+NAME                     STATUS    MESSAGE
+wanreplication-sample    Success
+----
+
+You can also monitor the status of the WAN replications in Management Center dashboard. Learn to deploy Management Center at xref:get-started.adoc#step-4-start-management-center[Start Management Center].
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/wan-replication.adoc:67]:
```suggestion
You can also monitor the status of the WAN replications in xref:get-started.adoc#step-4-start-management-center[Management Center].
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Hi there, Just added comments for consistency and tried to suggest where users might need more information. Apologies if I have misunderstood/misinterpreted any information. Please do let me know if anything doesn't make sense.

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
+= Configuring WAN Replication
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply docs/modules/ROOT/pages/wan-replication.adoc:1]: Humm, I am not sure. We did not mention persistence was an enterprise feature. Maybe we can add to both.

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+|Queue is the configuration for WAN events queue.
+
+- capacity: Capacity is the total capacity of WAN queue.
+- fullBehavior: FullBehavior represents the behavior of the new arrival when the queue is full.
+
+|batch
+|Batch is the configuration for WAN events batch.
+
+- size: Size represents the maximum batch size.
+- maximumDelay: MaximumDelay represents the maximum delay in milliseconds. If the batch size is not reached, the events will be sent after the maximum delay.
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply docs/modules/ROOT/pages/wan-replication.adoc:35]: Default is shown in the API Reference.

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+|targetClusterName
+|targetClusterName is the cluster name of the target Hazelcast cluster.
+
+|endpoints
+|Endpoints is the comma separated target cluster member adresses.
+
+|queue
+|Queue is the configuration for WAN events queue.
+
+- capacity: Capacity is the total capacity of WAN queue.
```
[the remark this answers is not in the document]
Comment (mtyazici) [reply docs/modules/ROOT/pages/wan-replication.adoc:28]: Default is shown in the API Reference

File: docs/modules/ROOT/pages/wan-replication.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (angelasimms) [reply docs/modules/ROOT/pages/wan-replication.adoc:1]: Hi there, As it seems like licensing is required for both features, I think it makes sense. It looks like you just add the following to the top of the page `:page-enterprise: true` to add the enterprise tag.

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
+= Configuring WAN Replication
+:description: pass:q[WAN Replication allows you to keep Hazelcast clusters in sync across multiple sites or cloud deployments. In Hazelcast Platform Operator, you can create a `WanReplication` resource to replicate a map from the source to the target Hazelcast cluster. ]
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/wan-replication.adoc:2]: Sorry, this is my mistake, just a minor amend.

```suggestion
:description: pass:q[WAN Replication allows you to keep Hazelcast clusters in sync across multiple sites or cloud deployments. In Hazelcast Platform Operator, you can create a `WanReplication` resource to replicate a map from a source to a target Hazelcast cluster. ]
```

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+See xref:hazelcast:wan:wan.adoc[WAN replication] for more detailed information about the configuration and behavior of WAN replication in Hazelcast.
+
+== Prerequisites for WAN Replication
+
+Before you create and configure a `WanReplication` resource, you need to create the following:
+
+- A `Hazelcast` resource and `Map` resource for the source Hazelcast cluster.
+- A target Hazelcast cluster that is accessible from the source cluster.                
+== Configuring the WanReplication Resource
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/wan-replication.adoc:15]: Just needs an extra space before the heading.
```suggestion

== Configuring the WanReplication Resource
```

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+
+You can use the following configuration options in your `WanReplication` resource. For more detailed information, see the xref:api-ref.adoc#wanreplicationspec[API Reference].
+
+[cols="20%m,80%a"]
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/wan-replication.adoc:20]: The field name on the second row of the table is wrapping so maybe we could adjust the column sizes to `[cols="25%m,75%a"]`?

File: docs/modules/ROOT/pages/wan-replication.adoc [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+|Configuration of the acknowledgement sent after a batch of replication events is received by the target cluster:
+
+- `type`: Whether an acknowledgement is sent as soon as the batch of events is received (`ACK_ON_RECEIPT`) or after each update is applied to the target cluster (`ACK_ON_OPERATION_COMPLETE`).
+- `timeout`: The period of time the source cluster waits for the acknowledgement. If the confirmation is not received before the timeout, the replication events are resent.
+
+|===
+
+== Example Configuration
+
+The following `WanReplication` resource creates the WAN Replication configuration for the `Hazelcast` and `Map` resources on the source Hazelcast cluster.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/wan-replication.adoc:55]:
```suggestion
The following `WanReplication` resource creates the WAN replication configuration for the `Hazelcast` and `Map` resources on the source Hazelcast cluster.
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
Just some really minor amends. Thanks for doing the updates so far.

File: docs/modules/ROOT/pages/wan-replication.adoc (hunk 7 above)
Comment (mtyazici) [reply docs/modules/ROOT/pages/wan-replication.adoc:20]: I will update it

File: docs/modules/ROOT/pages/wan-replication.adoc (hunk 6 above)
Comment (mtyazici) [reply docs/modules/ROOT/pages/wan-replication.adoc]: I already had done this change :+1:

**Review by angelasimms — APPROVED** [review]
All looks good to me. Thanks.

**Comment by mtyazici** [comment]
Thank you as well, I will merge the PR :)

**Pull request merged**